R/deploy-site.R

```diff
@@ -0,0 +1,91 @@
+#' Deploy a pkgdown site on Travis-CI to Github Pages
+#'
+#' `deploy_site_github()` sets up your SSH keys for deployment, builds the
+#' site with [build_site()], commits the site to the `gh-pages` branch and then pushes
+#' the results back to GitHub. `deploy_site_github()` is meant only to be used
+#' by the CI system on Travis, it should not be called locally. See 'Setup' for
+#' details on setting up your repository to use this.
+#'
+#' @section Setup:
+#' Add the following to your `.travis.yml` file.
+#'
+#' ```
+#' before_deploy: Rscript -e 'install.packages("pkgdown")'
+#' deploy:
+#'   provider: script
+#'   script: Rscript -e 'pkgdown::deploy_site_github()'
```

Comment:
Do we still need `skip-cleanup: true` in here?

Reply:
I've seen an instance in _tic_ where it is helpful. Without this setting, everything is stashed away before deployment, including e.g. results of `configure` scripts and perhaps `*.o` files. These files would be rebuilt.

Reply:
...perhaps not if _pkgdown_ installs from the built tarball, as I've seen in the source.

Reply:
I think you do need it, the tarball is still in the package working directory, sorry for the omission.